shell/plugins/dashboard-store/resource-class.js

```diff
@@ -851,7 +854,11 @@ export default class Resource {
 
   // You can add custom actions by overriding your own availableActions (and probably reading super._availableActions)
   get _availableActions() {
-    const all = [
+    // get menu actions available by plugins configuration
+    const currentRoute = this.currentRouter().app._route;
+    const extensionMenuActions = getApplicableExtensionEnhancements(this.$rootState, ExtensionPoint.ACTION, ActionLocation.TABLE, currentRoute, this);
```

Comment:
Comment only. This will run a LOT. getApplicableExtensionEnhancements and checkExtensionRouteBinding need to remain really efficient